src/EditorFeatures/CSharpTest/GenerateComparisonOperators/GenerateComparisonOperatorsTests.cs

```diff
@@ -0,0 +1,448 @@
+            => FlattenActions(actions);
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestClass()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+
+[||]class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+}",
+@"
+using System;
+
+class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right)
+    {
+        return left.CompareTo(right) < 0;
+    }
+
+    public static bool operator >(C left, C right)
+    {
+        return left.CompareTo(right) > 0;
+    }
+
+    public static bool operator <=(C left, C right)
+    {
+        return left.CompareTo(right) <= 0;
+    }
+
+    public static bool operator >=(C left, C right)
+    {
+        return left.CompareTo(right) >= 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestPreferExpressionBodies()
+        {
+            await TestInRegularAndScriptAsync(
+@"
+using System;
+
+[||]class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+}",
+@"
+using System;
+
+class C : IComparable<C>
+{
+    public int CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right) => left.CompareTo(right) < 0;
+    public static bool operator >(C left, C right) => left.CompareTo(right) > 0;
+    public static bool operator <=(C left, C right) => left.CompareTo(right) <= 0;
+    public static bool operator >=(C left, C right) => left.CompareTo(right) >= 0;
+}", options: Option(CSharpCodeStyleOptions.PreferExpressionBodiedMethods, CSharpCodeStyleOptions.WhenPossibleWithSuggestionEnforcement));
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestExplicitImpl()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+
+[||]class C : IComparable<C>
+{
+    int IComparable<C>.CompareTo(C c) => 0;
+}",
+@"
+using System;
+
+class C : IComparable<C>
+{
+    int IComparable<C>.CompareTo(C c) => 0;
+
+    public static bool operator <(C left, C right)
+    {
+        return ((IComparable<C>)left).CompareTo(right) < 0;
+    }
+
+    public static bool operator >(C left, C right)
+    {
+        return ((IComparable<C>)left).CompareTo(right) > 0;
+    }
+
+    public static bool operator <=(C left, C right)
+    {
+        return ((IComparable<C>)left).CompareTo(right) <= 0;
+    }
+
+    public static bool operator >=(C left, C right)
+    {
+        return ((IComparable<C>)left).CompareTo(right) >= 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateComparisonOperators)]
+        public async Task TestOnInterface()
+        {
+            await TestInRegularAndScript1Async(
+@"
+using System;
+
+class C : [||]IComparable<C>
```

Comment:
am I missing something or is this the same test as `TestClass`?

Reply:
scratch that, it's the cursor placement not the type inheritance.

Reply:
:)